### ISSUE TITLE
Documentations

### ISSUE THREAD

**Issue by JoseskVolpe** [faissaloo/telegram-micro#3]
As a open-source project, i think adding descriptions to the methods, and bibliography to your codes with diagrams, would help a lot. I saw it's only SSH stuff (wich i never messed with) until then and there should be some documentations explaining how that works.

**Comment by faissaloo** [comment]
I agree but there's alot so I need a place to start and an audience to write for. If you could tell me what would be most important for you to understand or what's most confusing that'd be very helpful.  
Also there's no code here that has anything to do with SSH, this application implements MTProto entirely on the device and connects to Telegram servers directly unlike other attempts to make J2ME clients I've seen.
